tests/routes/test_auth.py

```diff
@@ -2,113 +2,165 @@
+
+        response = self.client.get("/auth/callback")
+        self.assertEqual(response.status_code, 200)
+        mock_get_token.assert_called_once()
+        mock_process_user_session.assert_called_once()
+        mock_send_github_invitation.assert_called_once()
+        mock_render_success_page.assert_called_once()
+
+    @patch("join_github_app.main.routes.auth.render_error_page", return_value=Response(status=500))
+    @patch("join_github_app.main.routes.auth.get_token")
+    def test_callback_route_token_failure(self, mock_get_token, mock_render_error_page):
+        mock_get_token.side_effect = AuthTokenError("Token error")
+
+        response = self.client.get("/auth/callback")
+        self.assertEqual(response.status_code, 500)
+        mock_get_token.assert_called_once()
+        mock_render_error_page.assert_called_once()
+
+    @patch("join_github_app.main.routes.auth.ALLOWED_EMAIL_DOMAINS",
+           new=["example.com", "test.com"])
+    def test_process_user_session_success(self):
+        with self.app.test_request_context('/auth/callback'):
+            session['user'] = {'userinfo': {'email': 'user@test.com'}}
+            session['email'] = 'user@test.com'
+            session['org_selection'] = ['some_org']
+
+            result = process_user_session()
+            self.assertTrue(result)
+
+    def test_process_user_session_without_session(self):
+        with self.app.test_request_context('/auth/callback'):
+            session['user'] = {'userinfo': {'email': ''}}
+            session['email'] = ''
+            session['org_selection'] = ['']
+
+            result = process_user_session()
+            self.assertFalse(result)
+
+    def test_process_user_session_without_approved_email(self):
+        with self.app.test_request_context('/auth/callback'):
+            session['user'] = {'userinfo': {'email': 'user@test.com'}}
+            session['email'] = 'user@test.com'
+            session['org_selection'] = ['some_org']
+
+            result = process_user_session()
+            self.assertFalse(result)
+
+    @patch("join_github_app.main.routes.auth.ALLOWED_EMAIL_DOMAINS",
+           new=["example.com", "test.com"])
+    def test_process_user_session_with_invalid_user(self):
+        with self.app.test_request_context('/auth/callback'):
+            session['user'] = {'userinfo': {'email': 'user@test.com'}}
+            session['email'] = 'user@test2.com'
+            session['org_selection'] = ['some_org']
+
+            result = process_user_session()
+            self.assertFalse(result)
+
+    @patch("join_github_app.main.routes.auth.ALLOWED_EMAIL_DOMAINS",
+           new=["example.com", "test.com"])
+    def test_process_user_session_with_missing_org_selection(self):
+        with self.app.test_request_context('/auth/callback'):
+            session['user'] = {'userinfo': {'email': 'user@test.com'}}
+            session['email'] = 'user@test2.com'
+            session['org_selection'] = ['some_org']
```

Comment:
```suggestion
            session['org_selection'] = []
```
Missing org selection?